src/components/pages/dashboard-pages/Notifications.jsx

```diff
@@ -158,131 +166,140 @@ const Notifications = ({ socket }) => {
         New applications:
       </h1>}
       {classesError && <p className="text-red">{classesError.message}</p>}
-      <div className="flex flex-col gap-6 mt-10 m-5 md:m-10">
-        {classes.length > 0 ? (
-          classes.map((classInfo) => (
-            <div
-              key={classInfo._id}
-              className="border rounded-lg border-gray p-5 bg-pureWhite"
-            >
-              <h2 className="font-roboto font-medium text-2xl mb-4">
-                {classInfo.classTitle}
-              </h2>
-              <hr className="my-4 border-gray" />
-              {classInfo.applications.map((application) => {
-                const applicantDetails = applicants.find(
-                  (user) => user._id === application.userId
-                );
-                return (
-                  <div
-                    key={application._id}
-                    className="flex flex-row items-start mb-4 border-b border-gray pb-4"
-                  >
-                    <div className="flex w-1/4 justify-center items-center">
-                      <img
-                        className="rounded-lg w-30 h-35 "
-                        src={
-                          applicantDetails
-                            ? applicantDetails.profileImageUrl
-                            : '/default-profile.png'
-                        }
-                        alt={
-                          applicantDetails
+      {isLoading ? (
+        <Loader />
+      ) : (
+        <div className="flex flex-col gap-6 mt-10 m-5 md:m-10">
+          {classes.length > 0 ? (
+            classes.map((classInfo) => (
+              <div
+                key={classInfo._id}
+                className="border rounded-lg border-gray p-5 bg-pureWhite"
+              >
+                <h2 className="font-roboto font-medium text-2xl mb-4">
+                  {classInfo.classTitle}
+                </h2>
+                <hr className="my-4 border-gray" />
+                {classInfo.applications.map((application) => {
```

Comment:
in future to avoid long map in return you can move it into separate component or at least variable. Deeply nested logic is hard to read. Treat this comment as recommendation , not call to action

Reply:
Thank you, Anna